custom_reg_form/management/commands/email_opt_in_list_to_s3.py

```diff
@@ -0,0 +1,257 @@
+        """.format(
+            field_names=field_names, reg_ext_table=reg_ext_table
+        )
+
+    return query
+
+
+class Command(BaseCommand):
+    """
+    Admin command to export User/UserProfile/Registration Extension fields to csv file on s3
+    """
+
+    help = "Exports users email opt-in preferences to an s3 bucket"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--use-temp-file",
+            action="store_true",
+            help=(
+                "Use a temp file instead of keeping the whole csv in memory,"
+                "useful for when expecting a large data set"
+            ),
+        )
+        parser.add_argument(
+            "--skip-when-null",
+            dest="skip_when_null",
+            action="store_true",
+            help=(
+                "when enabled the script skips the row when the value of"
+                " one of the fields is null (default=True)"
+            )
+        )
+        parser.add_argument(
+            "--no-skip-when-null", dest="skip_when_null", action="store_false"
+        )
+        parser.set_defaults(skip_when_null=True)
+        parser.add_argument(
+            "--aws-access-key-setting",
+            type=str,
+            help="specify which setting to use as an aws access key",
+        )
+        parser.add_argument(
+            "--aws-secret-key-setting",
+            type=str,
+            help="specify which setting to use as an aws secret key",
+        )
+        parser.add_argument(
+            "--aws-profile", type=str, help="specify which aws profile to use"
+        )
+        parser.add_argument(
+            "--fields",
+            type=str,
+            nargs="+",
+            help="fields to include in the output, see docstring header for how to specify fields",
+        )
+        parser.add_argument(
+            "--usage", action="store_true", help="print the module docstring header"
+        )
+        parser.add_argument(
+            "--s3-bucket-name",
+            type=str,
+            help="aws s3 bucket name where the csv file will be saved.",
+        )
+        parser.add_argument(
+            "--s3-object-name",
+            type=str,
+            help="the name under which the output will be saved.",
+        )
+
+    def handle(self, *args, **options):
+
+        if options["usage"]:
+            print(__doc__)
+            return
+
+        # The csv writer and boto3 client require different file modes
+        binary_fd = (
+            tempfile.TemporaryFile(mode="w+b")
+            if options["use_temp_file"]
+            else io.BytesIO()
+        )
+        text_fd = io.TextIOWrapper(binary_fd, encoding="utf-8", write_through=True)
+
+        writer = csv.writer(text_fd)
+
+        cursor = connections[read_replica_or_default()].cursor()
+
+        query = construct_query(options["fields"])
+
+        cursor.execute(query)
+
+        while 1:
+            array = (
```

Comment:
I understand that it is an array, but what kind of array?

Reply:
Renamed to rows.